src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemService.Cancel.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Store.Entries;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+using Microsoft.Health.Dicom.Core.Messages.Workitem;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to process the list of <see cref="IDicomInstanceEntry"/>.
+    /// </summary>
+    public partial class WorkitemService
+    {
+        public async Task<CancelWorkitemResponse> ProcessCancelAsync(DicomDataset dataset, string workitemInstanceUid, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+
+            var workitemMetadata = await _workitemOrchestrator
+                .GetWorkitemMetadataAsync(workitemInstanceUid, cancellationToken)
```

Comment:
I know we are throwing argument null exception if the metadata is not exists, but dont we need to throw workitem not found exception?